src/vmm/src/seccomp_filters/mod.rs

```diff
@@ -46,6 +52,89 @@ impl fmt::Display for FilterError {
     }
 }
 
+/// Seccomp filter configuration.
+pub enum SeccompConfig {
+    /// Seccomp filtering disabled.
+    None,
+    /// Basic filtering, matching only on syscall numbers.
+    Basic,
+    /// Default, advanced filters.
+    /// Checks both syscall numbers and argument values (where applicable).
+    Advanced,
+    /// Custom, user-provided filters.
+    Custom(Box<dyn std::io::Read>),
+}
+
+impl SeccompConfig {
+    /// Given the relevant command line args, return the appropriate config type.
+    pub fn from_args(
+        seccomp_level: Option<&String>,
+        no_seccomp: bool,
+        seccomp_filter: Option<&String>,
+    ) -> Result<Self, FilterError> {
+        // The argument parser is configured to forbid usages of `--seccomp-filter` or `--no-seccomp`
+        // together with `--seccomp-level`, so we do not have to check for it.
+        match seccomp_level {
+            Some(value) => match &value[..] {
+                "0" => Ok(SeccompConfig::None),
+                "1" => Ok(SeccompConfig::Basic),
+                "2" => Ok(SeccompConfig::Advanced),
+                _ => Err(FilterError::SeccompConfig(
+                    "Invalid value for --seccomp-level.".to_string(),
+                )),
+            },
+            None => match no_seccomp {
+                true => Ok(SeccompConfig::None),
+                false => match seccomp_filter {
+                    Some(path) => Ok(SeccompConfig::Custom(Box::new(
+                        File::open(&path).map_err(FilterError::FileOpen)?,
+                    ))),
+                    None => Ok(SeccompConfig::Advanced),
+                },
+            },
+        }
+    }
+}
+
+/// Retrieve the appropriate filters, based on the SeccompConfig.
+pub fn get_filters(config: SeccompConfig) -> Result<BpfThreadMap, FilterError> {
+    match config {
+        SeccompConfig::None => Ok(get_empty_filters()),
+        SeccompConfig::Basic => get_default_filters(true),
+        SeccompConfig::Advanced => get_default_filters(false),
+        SeccompConfig::Custom(reader) => get_custom_filters(reader),
+    }
+}
+
+/// Retrieve the default filters containing the syscall rules required by `Firecracker`
+/// to function. The binary file is generated via the `build.rs` script of this crate.
+fn get_default_filters(basic: bool) -> Result<BpfThreadMap, FilterError> {
+    // Retrieve, at compile-time, the serialized binary filter generated with seccompiler.
+    let bytes: &[u8] = match basic {
+        true => include_bytes!(concat!(env!("OUT_DIR"), "/basic_seccomp_filter.bpf")),
+        false => include_bytes!(concat!(env!("OUT_DIR"), "/seccomp_filter.bpf")),
```

Comment:
Nit: I think it would be easier for future maintenance to have the names of these files stored in const variables.

Reply:
It would be nice. However, the `concat!` macro only works on string literals. See this issue: https://github.com/rust-lang/rust/issues/31383